tox.ini

```diff
@@ -927,52 +927,79 @@ commands =
     pytest
 
 [testenv:dev.k8s.registry]
-changedir = {toxinidir}/packages/grid
+description = Start local Kubernetes registry with k3d
+changedir = {toxinidir}
 passenv=HOME,USER
 allowlist_externals =
     bash
+    sudo
 commands =
-    bash -c 'k3d registry create registry.localhost --port 12345 -v $HOME/.k3d-registry:/var/lib/registry'
-    bash -c '\
-        CYAN="\033[1;36m"; \
-        CLEAR="\033[0m"; \
-        printf "\n${CYAN}---------------------------------------------------------\n" ;\
-        printf "Make sure to add the following to your /etc/hosts file:${CLEAR}\n";\
-        printf "127.0.0.1 k3d-registry.localhost\n";\
-        printf "${CYAN}---------------------------------------------------------${CLEAR}\n"'
+    bash -c 'k3d registry create registry.localhost --port 12345 -v $HOME/.k3d-registry:/var/lib/registry || true'
+    sudo {envpython} scripts/patch_hosts.py --add-k3d-registry --fix-docker-hosts
```

Comment:
Will `sudo` work from powershell / cmd.exe on windows?

Reply:
Not as of now - I'll have a fix in a separate PR

Reply:
Logged an Issue for future: https://github.com/OpenMined/Heartbeat/issues/833